### PR TITLE
Install File Browser Extension Packages

### PR DESCRIPTION
**Pull request by mmattel**
Fixes: #267 ([AppImage] HowTo install file browser extension packages manually)

Describe how to install File Browser Extension Packages.

Backport to 3.0 and 2.11

### CODE REVIEW

File: modules/ROOT/pages/installing.adoc [hunk 1]
```diff
@@ -204,6 +205,17 @@ sudo chmod +x ownCloud-2.10.1.7187.AppImage
...
+==== File Browser Extension Packages
+
+ownCloud supports installing shell / file browser extensions, which means overlay icons and the context menu, via a bash script guided installation for Linux based systems. This script covers main systems and file browsers, but by nature not all possible flavours and combinations. Beside an auto mode, you can also manually define parameters from a list. The following quick command installing the extensions requires `curl` to be available on your system:
+
+[source,bash]
+----
+curl -s https://raw.githubusercontent.com/owncloud/client-desktop-install-shell-integration/main/install-extensions.sh | bash -s - --auto
+----
+
+If curl ist not available or you are not sure or want to read more, see {install-shell-integration-url}[Install ownCloud Linux Shell Extensions,window=_blank] for more details.
```
Comment (phil-davis) [inline modules/ROOT/pages/installing.adoc:217]:
```suggestion
If curl is not available or you are not sure or want to read more, see {install-shell-integration-url}[Install ownCloud Linux Shell Extensions,window=_blank] for more details.
```

**Comment by phil-davis** [comment]
I have Ubuntu 20.04 and ownCloud client:
```
ownCloud 3.0.0.9215- [457b08](https://github.com/owncloud/client/commit/457b08ed362e4c9ae430797508791ff25171c36c)
Libraries Qt 5.15.2, OpenSSL 1.1.1f 31 Mar 2020
Using virtual files plugin: suffix
ubuntu-5.4.0-135-generic
```

When trying the script I get:
```
$ curl -s https://raw.githubusercontent.com/owncloud/client-desktop-install-shell-integration/main/install-extensions.sh | bash -s - --auto
Looking for extension package for file browser nautilus on distro ubuntu...
Installing package nautilus-owncloud...
+ install_package nautilus-owncloud
+ sudo apt install nautilus-owncloud
[sudo] password for phil: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 nautilus-owncloud : Depends: owncloud-client (< 2.5.1.10973+dfsg-1ubuntu4.1~) but 3.0.0+oc-9215 is to be installed
E: Unable to correct problems, you have held broken packages.
Error: installation canceled
```

I have had desktop client 2.* on this system for a long time, and it has auto-updated recently to the desktop client 3 release.

Is there something that I need to do about `nautilus-owncloud` ?

File: modules/ROOT/pages/installing.adoc [hunk 2]
```diff
@@ -204,6 +205,17 @@ sudo chmod +x ownCloud-2.10.1.7187.AppImage
 * Note when you start the AppImage after setting it to be executable, AppImageLauncher will open if installed.
 --
 
+==== File Browser Extension Packages
+
+ownCloud supports installing shell / file browser extensions, which means overlay icons and the context menu, via a bash script guided installation for Linux based systems. This script covers main systems and file browsers, but by nature not all possible flavours and combinations. Beside an auto mode, you can also manually define parameters from a list. The following quick command installing the extensions requires `curl` to be available on your system:
```
Comment (EParzefall) [inline modules/ROOT/pages/installing.adoc:210]:
```suggestion
ownCloud supports installing shell / file browser extensions, which means overlay icons and the context menu, via a bash-script-guided installation for Linux-based systems. This script covers main systems and file browsers, but by nature not all possible flavors and combinations. Beside an auto mode, you can also manually define parameters from a list. The following quick command installing the extensions requires `curl` to be available on your system:
```

**Review by EParzefall — APPROVED** [review]

**Comment by mmattel** [comment]
> Is there something that I need to do about `nautilus-owncloud` ?

@fmoc can you have a look?

**Comment by fmoc** [comment]
@phil-davis unfortunately the (so far released) upstream Debian packages contain a problematic dependency on the desktop client package, see https://github.com/owncloud/client/issues/9034.

I can improve the script to try to detect such situations and in this case enforce the installation of our upstream packages.

**Comment by mmattel** [comment]
> I can improve the script

Let me know when done, I will then merge and initiate backporting. Thanks for taking care 😃 👍

File: modules/ROOT/pages/installing.adoc [hunk 3]
```diff
@@ -204,6 +205,17 @@ sudo chmod +x ownCloud-2.10.1.7187.AppImage
 * Note when you start the AppImage after setting it to be executable, AppImageLauncher will open if installed.
 --
 
+==== File Browser Extension Packages
+
+ownCloud supports installing shell / file browser extensions, which means overlay icons and the context menu, via a bash-script-guided installation for Linux-based systems. This script covers main systems and file browsers, but by nature not all possible flavors and combinations. Beside an auto mode, you can also manually define parameters from a list. The following quick command installing the extensions requires `curl` to be available on your system:
```
Comment (fmoc) [inline modules/ROOT/pages/installing.adoc:210]: We should use one term only. I think so far we've called them shell integration, not shell or file browser extensions.

How about something like this:
```suggestion
ownCloud supports installing shell integration packages, which allow you to add overlay icons and a special context menu to your file browsers, via a bash-script-guided installation for Linux-based systems. This script covers main systems and file browsers, but by nature not all possible flavors and combinations. Beside an auto mode, you can also manually define parameters from a list. The following quick command installing the extensions requires `curl` to be available on your system:
```

Comment (mmattel) [reply modules/ROOT/pages/installing.adoc]: That suggestion needed fixing on more locations - done

**Comment by mmattel** [comment]
@fmoc one thing is coming up: the command used (curl or wget) need sudo as you normally do not operate in su mode... I guess we need to change that here but also in the referenced repo.

**Review by fmoc — APPROVED** [review]

**Comment by fmoc** [comment]
The script will soon be able handle the situation interactively, so no need to change the docs.

**Comment by TheOneRing** [comment]
The current script does not install the shell extensions on one of the biggest distros out there.
In my opinion there is no harm in installing the outdated client, as long as we describe why we do it, and that the user should only use the Appimage and not the outdated 2.5 client.

**Comment by mmattel** [comment]
@TheOneRing 
* I do not get the context. If there is something missing in the script, pls file an own issue so @fmoc can track and fix it
* regarding installing an outdated client, again, have a new issue and point where this belongs to. I cant get the context here too

This PR is closed and anything writteh here can get lost / overseen easily...

**Comment by fmoc** [comment]
The script just forces users to add our repository on Debian/Ubuntu. It doesn't use the downstream packages if they depend on the client.

Please open issues in the new repository if you wish for changes.

**Pull request merged**